signac/contrib/project.py

```diff
@@ -1370,7 +1370,7 @@ def _read_cache(self):
     def temporary_project(self, dir=None):
         """Context manager for the initialization of a temporary project.
 
-        The temporary project is by default created within the root project's
+        The temporary project is by default created within the parent project's
```

Comment:
Parent project doesn't sound like the same as "root" in this context. To me, "parent" sounds like a higher level then the project.

Reply:
Well... it is sort of a parent. The temporary project is created in a subdirectory of the project instance calling this. This is awkward almost any way I can think of writing it, since it's hard to explain the benefit of this implementation detail.

Reply:
Proposal to also rewrite the beginning of the docstring to be:
```python
...
"""Context manager that initializes a temporary project within the calling Project.

The temporary project path defaults to that of the calling project workspace.

...

"""
```

btw - we should rename the argument from `dir` to `path`.

Reply:
This PR is focused on changing `root` to `path`. A follow-up PR to address this would be appropriate.